pkg/virt-controller/watch/vm_test.go

```diff
@@ -903,6 +903,48 @@ var _ = Describe("VirtualMachine", func() {
 			controller.Execute()
 		})
 
+		It("should detect that a DataVolume already exists and adopt it", func() {
+			vm, _ := DefaultVirtualMachine(false)
+			vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
+				Name: "test1",
+				VolumeSource: v1.VolumeSource{
+					DataVolume: &v1.DataVolumeSource{
+						Name: "dv1",
+					},
+				},
+			})
+
+			vm.Spec.DataVolumeTemplates = append(vm.Spec.DataVolumeTemplates, v1.DataVolumeTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "dv1",
+					Namespace: vm.Namespace,
+				},
+			})
+
+			addVirtualMachine(vm)
+
+			dv := createDataVolumeManifest(&vm.Spec.DataVolumeTemplates[0], vm)
+			dv.Status.Phase = cdiv1.Succeeded
+
+			orphanDV := dv.DeepCopy()
+			orphanDV.ObjectMeta.OwnerReferences = nil
+			dataVolumeInformer.GetStore().Add(orphanDV)
+
+			cdiClient.Fake.PrependReactor("patch", "datavolumes", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				patch, ok := action.(testing.PatchAction)
+				Expect(ok).To(BeTrue())
+				Expect(patch.GetName()).To(Equal(dv.Name))
+				Expect(patch.GetNamespace()).To(Equal(dv.Namespace))
+				Expect(string(patch.GetPatch())).To(ContainSubstring(string(vm.UID)))
```

Comment:
Nit: can you maybe also add a check that the "ownerReferences" key exists in the patch bytes ?

Reply:
I was actually looking for a programmatic way to actually apply the patch bytes to the DV object in memory, and then just inspect its `.ObjectMeta.OwnerReferences`. That turned out to be too complex, so will settle on substrings lookup.

Reply:
Done.